can/interfaces/socketcand/socketcand.py

```diff
@@ -40,11 +46,15 @@ def convert_can_message_to_ascii_message(can_message: can.Message) -> str:
     # Note: socketcan bus adds extended flag, remote_frame_flag & error_flag to id
     # not sure if that is necessary here
     can_id = can_message.arbitration_id
+    if can_message.is_extended_id:
+        can_id_string = f"{(can_id&0x1FFFFFFF):08X}"
+    else:
+        can_id_string = f"{(can_id&0x7FF):03X}"
     # Note: seems like we cannot add CANFD_BRS (bitrate_switch) and CANFD_ESI (error_state_indicator) flags
     data = can_message.data
     length = can_message.dlc
     bytes_string = " ".join(f"{x:x}" for x in data[0:length])
```

Comment:
@hartkopp Shouldn't the hex bytes here be zero-padded, e.g. `0c` instead of `c`?
If so, the format string should be `f"{x:02x}"`

Reply:
Doesn't `f"{(can_id&0x1FFFFFFF):08X}"` mean it is zero padded with a total length of 8 -> `00000123` ??
Because  this is what would be needed here.

`f"{x:02x}"` would be the right approach for data bytes (not CAN identifier), right?

Reply:
I'm talking about the data bytes here:
```python3
bytes_string = " ".join(f"{x:x}" for x in data[0:length])
```

For example, should it be `2 4 6 8 a` or `02 04 06 08 0a`? Does it make a difference for socketcand? Or is it okay as long as there is spacing between the bytes?

Reply:
It's supposed to be lowercase unpadded (%hhx):
https://github.com/linux-can/socketcand/blob/master/state_raw.c#L158-L171

Reply:
And on the receive side (< frame > message), the data is padded uppercase (%02X) with no spaces in between:
https://github.com/linux-can/socketcand/blob/master/state_raw.c#L125-L135

Reply:
Ouch! I wasn't aware of this inconsistency.
But I assume changing this in socketcand is not the way to go as it would potentially break other applications.

Reply:
Alright, then i will merge this.

Reply:
> Ouch! I wasn't aware of this inconsistency.
> But I assume changing this in socketcand is not the way to go as it would potentially break other applications.

Sent PR to correct the docs in socketcand:

https://github.com/linux-can/socketcand/pull/27